comms/discovery/server/server.go

```diff
@@ -31,6 +33,50 @@ func getHealthStatus() schema.Health {
 	}
 }
 
+func chatWebsocket(c echo.Context) error {
+	ctx := c.Request().Context()
+
+	// Check that timestamp is less than 5 seconds old
+	timestamp, err := strconv.ParseInt(c.QueryParam("timestamp"), 0, 64)
+	if err != nil || time.Now().UnixMilli() - timestamp > 5000 {
+		return c.String(400, "Invalid signature timestamp")
+	}
+	
+	// Websockets from the client can't send headers, so instead, the signature is a query parameter
+	// Strip out the signature query parameter to get the true signature payload
+	u, err := url.Parse(c.Request().RequestURI)
+	if err != nil {
+		return c.String(400, "Could not parse URL")
+	}
+	q := u.Query()
+	q.Del("signature")
+	u.RawQuery = q.Encode()
+	signature := c.QueryParam("signature")
+	signedData := []byte(u.String())
```

Comment:
wonder if we could do some standard HMAC thing here as this looks a lot like HMAC code.

I'll look into that a bit.

Reply:
Would love to standardize and use HMAC 🙏 How would we do this though without sharing the secret with the server?

Maybe we could have the DN and sender use a shared secret so that the DN doesn't need the client's secret?